### PR TITLE
cpu: remove superfluous definitions in ST header

### PR DESCRIPTION
**Pull request by OlegHahm**

### CODE REVIEW

**Comment by OlegHahm** [comment]
split up in two commits

**Comment by haukepetersen** [comment]
ACK when Travis...

File: cpu/stm32f1/periph/spi.c [hunk 1]
```diff
@@ -207,11 +207,11 @@ int spi_transfer_byte(spi_t dev, char out, char *in)
             return -1;
     }
 
-    while ((spi->SR & SPI_SR_TXE) == RESET);
+    while ((spi->SR & SPI_SR_TXE));
```
Comment (thomaseichinger) [inline cpu/stm32f1/periph/spi.c]: Missing the negation `!`.

File: cpu/stm32f1/periph/spi.c [hunk 2]
```diff
@@ -207,11 +207,11 @@ int spi_transfer_byte(spi_t dev, char out, char *in)
...
     spi->DR = out;
     transferred++;
 
-    while ((spi->SR & SPI_SR_RXNE) == RESET);
+    while ((spi->SR & SPI_SR_RXNE));
```
Comment (thomaseichinger) [inline cpu/stm32f1/periph/spi.c]: As above.

File: cpu/stm32f1/periph/spi.c (hunk 1 above)
Comment (OlegHahm) [inline cpu/stm32f1/periph/spi.c]: very true - thanks for being so cautious

**Comment by haukepetersen** [comment]
re-ACK. Please squash

**Comment by OlegHahm** [comment]
squashed

**Pull request merged**